Assets/Mirror/Editor/Weaver/Writers.cs

```diff
@@ -51,6 +52,73 @@ public static void Init(AssemblyDefinition CurrentAssembly)
                 { Weaver.transformType.FullName, Resolvers.ResolveMethodWithArg(networkWriterType, CurrentAssembly, "Write", Weaver.transformType) },
                 { "System.Byte[]", Resolvers.ResolveMethodWithArg(networkWriterType, CurrentAssembly, "WriteBytesAndSize", "System.Byte[]") }
             };
+
+            customWriters = new Dictionary<string, MethodReference>();
+            RegisterCustomWriters(CurrentAssembly);
+        }
+
+        static void RegisterCustomWriters(AssemblyDefinition currentAssembly)
+        {
+            foreach (TypeDefinition td in currentAssembly.MainModule.Types)
+            {
+                if (td.IsClass && td.BaseType.CanBeResolved())
+                {
+                    RegisterCustomWriters(td);
+                }
+            }
+        }
+
+        static void RegisterCustomWriters(TypeDefinition td)
+        {
+            foreach (MethodDefinition md in td.Methods)
+            {
+                foreach (CustomAttribute ca in md.CustomAttributes)
+                {
+                    if (ca.AttributeType.FullName == Weaver.WriterType.FullName)
+                    {
+                        RegisterCustomWriter(md);
+                        break;
+                    }
+                }
+            }
+
+            foreach (TypeDefinition embedded in td.NestedTypes)
+            {
+                RegisterCustomWriters(embedded);
+            }
+        }
+
+        private static void RegisterCustomWriter(MethodDefinition md)
+        {
+            if (md.Parameters.Count != 2)
+            {
+                Weaver.Error($"Writer {md.FullName} must receive a NetworkWriter and your parameter");
+                return;
+            }
+
+            ParameterDefinition writerParameter = md.Parameters[0];
+
+            if (writerParameter.ParameterType.Resolve().FullName != Weaver.NetworkWriterType.FullName)
+            {
+                Weaver.Error($"Writer {md.FullName} must receive a NetworkWriter");
```

Comment:
How about `"Writer {md.FullName} must have a NetworkWriter as the first parameter. "` to make it clear which parameter it must be?